engine/src/main/resources/swagger/cromwell.yaml

```diff
@@ -716,6 +756,208 @@ definitions:
+            valueType:
+              typeName: Array
+              arrayType:
+                typeName: String
+            optional: false
+          - name: my_wf.optional_input
+            valueType:
+              typeName: Optional
+              optionalType:
+                typeName: String
+            optional: true
+            default: "hello"
+          - name: my_wf.map_input
+            valueType:
+              typeName: Map
+              mapType:
+                keyType:
+                  typeName: String
+                valueType:
+                  typeName: Int
+            optional: false
+          - name: my_wf.object_input
+            valueType:
+              typeName: Object
+              objectFieldTypes:
+                - fieldName: "int_field"
+                  fieldType:
+                    typeName: Int
+                - fieldName: "int_array_field"
+                  fieldType:
+                    typeName: Array
+                    arrayType:
+                      typeName: Int
+            optional: false
+          - name: my_wf.int_string_pair_input
+            valueType:
+              typeName: Tuple
+              tupleTypes:
+                - typeName: Int
+                - typeName: String
+            optional: false
+      images:
+        type: array
+        items:
+          type: string
+          example:
+          - quay.io/seqware/seqware_full/1.1
+          - ubuntu:latest
+          - EXPRESSION_BASED_IMAGE
+        description: >-
+          All known docker images (and versions/hashes) used by this tool.
+          If the tool will have to evaluate one or more docker image strings at runtime, the placeholder
+          "EXPRESSION_BASED_IMAGE" will be used.
+      submitted_descriptor_type:
+        $ref: '#/definitions/DescriptorTypeAndVersion'
+      imported_descriptor_types:
+        type: array
+        items:
+          $ref: '#/definitions/DescriptorTypeAndVersion'
+        description: The set of all types imported by the submitted workflow.
+      meta:
+        type: object
+        description: An open set of key/value pairs echoing any workflow-level metadata found in the description file.
+        example:
+          author: "Batman"
+          copyright: "BSD 3-clause"
+  ToolParameter:
+    type: object
+    description: >-
+      An input parameter for a tool or workflow.
+    required:
+    - name
+    - valueType
+    - optional
+    properties:
+      name:
+        type: string
+        description: The name of this input value (formatted as expected by the tool)
+      valueType:
+        $ref: '#/definitions/ValueType'
+      optional:
+        type: boolean
+        description: Whether the tool allows this value to not be specified
+      default:
+        type: string
+        description: The in-language expression used to evaluate a default value for this parameter, if
+          none is supplied.
+      documentation:
```

Comment:
Is documentation essentially a first-class key of parameter_meta?

Reply:
ToL: There are certain types of workflows that we would want to feature and I would think we would expect all non-optional inputs to those workflows to require documentation. I'm thinking the groundwork laid here will help us head towards a distant future where one can "validate" a workflow such that it requires documentation.